src/ai/backend/gateway/domainconfig.py

```diff
@@ -0,0 +1,194 @@
+import logging
+import re
+from typing import Any, Tuple
+
+from aiohttp import web
+import aiohttp_cors
+import trafaret as t
+
+from ai.backend.common import msgpack
+from ai.backend.common.logging import BraceStyleAdapter
+
+from .auth import auth_required
+from .exceptions import (
+    InvalidAPIParameters, DotfileCreationFailed,
+    DotfileNotFound, DotfileAlreadyExists,
+    GenericForbidden, DomainNotFound
+)
+from .manager import READ_ALLOWED, server_status_required
+from .types import CORSOptions, Iterable, WebMiddleware
+from .utils import check_api_params
+
+from ..manager.models import (
+    domains,
+    query_domain_dotfiles,
+    verify_dotfile_name,
+    MAXIMUM_DOTFILE_SIZE,
+)
+
+log = BraceStyleAdapter(logging.getLogger('ai.backend.gateway.dotfile'))
+
+
+@server_status_required(READ_ALLOWED)
+@auth_required
+@check_api_params(t.Dict(
+    {
+        t.Key('domain'): t.String,
+        t.Key('data'): t.String(max_length=MAXIMUM_DOTFILE_SIZE),
+        t.Key('path'): t.String,
+        t.Key('permission'): t.Regexp(r'^[0-7]{3}$', re.ASCII),
+    }
+))
+async def create(request: web.Request, params: Any) -> web.Response:
+    log.info('CREATE DOTFILE (domain: {0})', params['domain'])
+    if not request['is_superadmin'] and not request['is_admin']:
+        raise GenericForbidden('Only superadmin or domain admin can create dotfiles')
+    elif request['is_admin'] and request['user']['domain_name'] != params['domain']:
+        raise GenericForbidden('Domain-admins cannot create dotfiles of other domains')
+
+    dbpool = request.app['dbpool']
+    async with dbpool.acquire() as conn, conn.begin():
+        dotfiles, leftover_space = await query_domain_dotfiles(conn, params['domain'])
+        if dotfiles is None:
+            raise DomainNotFound('Input domain is not found')
+        if leftover_space == 0:
+            raise DotfileCreationFailed('No leftover space for dotfile storage')
+        if len(dotfiles) == 100:
+            raise DotfileCreationFailed('Dotfile creation limit reached')
+        if not verify_dotfile_name(params['path']):
+            raise InvalidAPIParameters('dotfile path is reserved for internal operations.')
+
+        duplicate = [x for x in dotfiles if x['path'] == params['path']]
+        if len(duplicate) > 0:
+            raise DotfileAlreadyExists
+        new_dotfiles = list(dotfiles)
+        new_dotfiles.append({'path': params['path'], 'perm': params['permission'],
+                             'data': params['data']})
+        dotfile_packed = msgpack.packb(new_dotfiles)
+        if len(dotfile_packed) > MAXIMUM_DOTFILE_SIZE:
+            raise DotfileCreationFailed('No leftover space for dotfile storage')
+
+        query = (domains.update()
+                        .values(dotfiles=dotfile_packed)
+                        .where(domains.c.name == params['domain']))
+        await conn.execute(query)
+    return web.json_response({})
+
+
+@auth_required
```

Comment:
I think it would be better for domain dotfiles not to be queried by domain admins and users in the other domains.